include/openPMD/auxiliary/Option.hpp

```diff
@@ -0,0 +1,177 @@
+
+#if __cplusplus >= 201703L
+#   include <variant> // IWYU pragma: export
+namespace variantSrc = std;
+#else
+#   include <mpark/variant.hpp> // IWYU pragma: export
+namespace variantSrc = mpark;
+#endif
+
+namespace openPMD
+{
+namespace auxiliary
+{
+    namespace detail
+    {
+        struct Empty
+        {
+        };
+    } // namespace detail
+
+    /**
+     * @brief Simple Option type based on variantSrc::variant.
+     *
+     * @tparam T Type that can be optionally stored in an Optional object.
+     */
+    template< typename T >
+    class Option
+    {
+        using data_t = variantSrc::variant< T, detail::Empty >;
+        data_t m_data;
+
+    public:
+        /**
+         * @brief Create an empty Option.
+         *
+         */
+        explicit Option() : m_data( detail::Empty() )
+        {
+        }
+
+        /**
+         * @brief Create a full Option.
+         *
+         * @param data The object to emplace in the Option.
+         */
+        Option( T data ) : m_data( std::move( data ) )
+        {
+        }
+
+        Option( Option const & other ) = default;
+
+        Option &
+        operator=( Option && other )
+        {
+            if( other.has_value() )
+            {
+                m_data.template emplace< 0 >( std::move( other.get() ) );
+            }
+            else
+            {
+                m_data.template emplace< 1 >( detail::Empty() );
+            }
+            return *this;
+        }
+
+        Option &
+        operator=( Option const & other )
+        {
+            if( other.has_value() )
+            {
+                m_data.template emplace< 0 >( other.get() );
+            }
+            else
+            {
+                m_data.template emplace< 1 >( detail::Empty() );
+            }
+            return *this;
+        }
+
+        bool
+        operator==( Option const & other ) const
+        {
+            if( has_value() )
+            {
+                return !other.has_value();
+            }
+            else
+            {
+                if( !other.has_value() )
+                {
+                    return false;
+                }
+                else
+                {
+                    return get() == other.get();
+                }
+            }
+        }
+
+        bool
+        operator!=( Option const & other ) const
+        {
+            return !( *this == other );
+        }
+
+        /**
+         * @return Is an object constantly stored in this?
+         */
+        bool
+        has_value() const
+        {
+            return m_data.index() == 0;
+        }
+
+        /**
+         * @return Is an object constantly stored in this?
+         */
+        operator bool() const
+        {
+            return has_value();
+        }
+
+        /**
+         * @brief Access the emplaced object if one is present.
+         *
+         * @throw std::bad_variant_access if no object is present.
+         * @return T const& The emplaced object.
```

Comment:
The type is contained and picked up from the signature:
```suggestion
         * @return The emplaced object.
```